server/src/main/java/org/elasticsearch/search/DocValueFormat.java

```diff
@@ -208,6 +210,20 @@ public DateTime(StreamInput in) throws IOException {
                 this.timeZone = ZoneId.of(zoneId);
                 this.resolution = DateFieldMapper.Resolution.ofOrdinal(in.readVInt());
             }
+            if (in.getVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
CURRENT should not be used in serialization logic since it is constantly changing. This should be the specific version this will be backported to

Reply:
good point, agree